pkg/agent/run.go

```diff
@@ -42,6 +42,12 @@ var (
 func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 	nodeConfig := config.Get(ctx, cfg, proxy)
 
+	if cfg.Docker {
+		if err := os.Symlink("/var/run/dockershim.sock", "/run/k3s/containerd/containerd.sock"); err != nil {
+			return err
+		}
+	}
+
```

Comment:
IMO, if we are going to have side-effects, it would be better to instead drop an `/etc/crictl.yaml` with the location of the CRI endpoint regardless of backing implementation. See:
- https://kubernetes.io/docs/tasks/debug-application-cluster/crictl/#general-usage
- https://github.com/kubernetes-sigs/cri-tools/blob/master/docs/crictl.md